### PR TITLE
use Model.new instead of ctor

### PR DESCRIPTION
**Pull request by nitzantomer**

### CODE REVIEW

File: scripts/src/models/index.ts [hunk 1]
```diff
@@ -14,24 +15,39 @@ let connection: Connection;
...
-		this.id = generateId(prefix);
+		return map;
 	}
+
+	@PrimaryColumn()
+	public id: string;
 }
 
 export abstract class CreationDateModel extends Model {
+	protected static initializers = Model.copyInitializers({ createdDate: () => new Date() });
```
Comment (doodyparizada) [inline scripts/src/models/index.ts]: can a Column accept a default value as a function instead?

File: scripts/src/create.ts [hunk 2]
```diff
@@ -236,18 +236,36 @@ async function createApp(appId, apiKey, name) {
 	const jwtPublic = fs.readFileSync("./examples/jwt_public_key.pem", "utf-8");
```
Comment (doodyparizada) [inline scripts/src/create.ts]: here `orderFromOffer` there is `new Order()`

File: scripts/src/models/index.ts [hunk 3]
```diff
@@ -3,37 +3,77 @@ import { BaseEntity, Column, createConnection, PrimaryColumn, Connection, Connec
...
-		super(prefix);
-		this.createdDate = new Date();
-	}
 }
 
 export function register(ctor: ModelConstructor) {
 	entities.push(ctor);
 }
 
+export function initializer(propName: string, fn: ModelMemberInitializer) {
```
Comment (doodyparizada) [inline scripts/src/models/index.ts]: `return initializers({[propName]: fn})`

File: scripts/src/public/services/orders.ts [hunk 4]
```diff
@@ -106,7 +106,15 @@ export async function submitOrder(
 	}
 
 	// transition open order to pending order
-	const order = new db.Order(openOrder, offer);
+	const order = db.Order.new({
+		id: openOrder.id,
+		userId: openOrder.userId,
+		offerId: openOrder.offerId,
+		amount: offer.amount,
+		type: "earn",
```
Comment (doodyparizada) [inline scripts/src/public/services/orders.ts]: `		type: offer.type,`

File: scripts/src/models/index.ts (hunk 3 above)
Comment (nitzantomer) [reply scripts/src/models/index.ts]: 👍

**Pull request merged**